src/server/JSON/JSON.hpp

```diff
@@ -0,0 +1,367 @@
+	 */
+	JSON(const Array& arr);
+
+	/**
+	 * @brief Constructs a JSON value storing an array (move).
+	 * @param arr Array of JSON values to move into the new value.
+	 *
+	 * @details This saves an unnecessary copy if the array is temporary.
+	 */
+	JSON(Array&& arr);
+
+	/**
+	 * @brief Constructs a JSON value storing an object (copy).
+	 * @param obj Object of key-value pairs (copied into the new value).
+	 *
+	 * @details Allows construction of a JSON value representing an object using
+	 * an existing map of key-value pairs.
+	 */
+	JSON(const Object& obj);
+
+	/**
+	 * @brief Constructs a JSON value storing an object (move).
+	 * @param obj Object of key-value pairs to move into the new value.
+	 *
+	 * @details Enables efficient construction when the object is temporary.
+	 */
+	JSON(Object&& obj);
+
+public:
+	/**
+	 * @brief Retrieves the type of the stored JSON value.
+	 * @return A JSON::Type enumeration value indicating the current type.
+	 */
+	Type GetType() const;
+
+	/**
+	 * @brief Gets the stored boolean value.
+	 * @return The boolean value.
+	 * @throw std::runtime_error if the stored value is not a boolean.
+	 */
+	bool AsBool() const;
+
+	/**
+	 * @brief Gets the stored numeric value.
+	 * @return The numeric value stored as a double.
+	 * @throw std::runtime_error if the stored value is not a number.
+	 */
+	double AsNumber() const;
+
+	/**
+	 * @brief Gets the stored string value.
+	 * @return A constant reference to the string.
+	 * @throw std::runtime_error if the stored value is not a string.
+	 */
+	const std::string& AsString() const;
+
+	/**
+	 * @brief Gets the stored array.
+	 * @return A constant reference to the array.
+	 * @throw std::runtime_error if the stored value is not an array.
+	 */
+	const Array& AsArray() const;
+
+	/**
+	 * @brief Gets the stored object.
+	 * @return A constant reference to the object.
+	 * @throw std::runtime_error if the stored value is not an object.
+	 */
+	const Object& AsObject() const;
+
+public:
+	/**
+	 * @brief Gets the underlying variant holding the JSON value.
+	 * @return A constant reference to the variant.
+	 *
+	 * @details This is useful for low-level access or for implementing additional
+	 * functionality without exposing implementation details.
+	 */
+	const Value& GetValue() const;
+
+	/**
+	 * @brief Template method to retrieve the stored value as a specific type.
+	 * @tparam T The type to which the stored value should be converted.
+	 * @return The stored value cast to type T.
+	 * @throw std::runtime_error if the stored value cannot be converted to type T.
+	 *
+	 * @details This method is a convenient wrapper around std::get. It helps to ensure
+	 * that the correct type is retrieved from the underlying variant.
+	 */
+	template<typename T>
+	T Get() const
+	{
+		try
+		{
+			return std::get<T>(m_value);
+		}
+		catch (const std::bad_variant_access&)
+		{
+			throw std::runtime_error("JSON conversion error: type mismatch");
+		}
```

Comment:
I think it's better if we try to throw one exception instead of two:
```suggestion
                if (!std::holds_alternative<T>(m_value))
                {
                    throw std::runtime_error("JSON conversion error: type mismatch");
                }
                return std::get<T>(m_value);
```